### ISSUE TITLE
When wrapped Dropdown with Field, A11y insights checker report 'Ensure buttons have discernible text' issue

### ISSUE THREAD

**Issue by sunyd0112** [microsoft/fluentui#33391]
### Component

Field

### Package version

9.56.3 

### React version

18.3.1 

### Environment

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  npmPackages:
    @fluentui/react-components: ^9.0.0 => 9.56.3 
    @types/react: ^17 => 17.0.83 
    @types/react-dom: ^17 => 17.0.25 
    react: ^18 => 18.3.1 
    react-dom: ^18 => 18.3.1
```

### Current Behavior

When wrapping a Dropdown component with Field, A11y insights checker will report 'Ensure buttons have discernible text' issue.
This is because the Dropdown button wraps the button text with a <span> element together, so the text cannot be recognized correctly. 

And even though the Field label is associated with the button, seems like the 'for' attribute on the label to find the 'id' on the button is not recognized by A11y insights checker. 
On W3C website, I only find examples about this usage on input fields, but not on buttons. Maybe use 'aria-labelledby' on button is a better option.

### Expected Behavior

When wrapping Dropdown with Field, there should not be any A11y issue.

### Reproduction

https://stackblitz.com/edit/n5f9lc?file=src%2Fexample.tsx

### Steps to reproduce

1. Go to the example link. Wait for the preview to load.
2. Open the accessibility insights checker and click 'Fast pass'. If you don't have this extension, maybe you need to download in Edge extension store first.
3. Observe the issue. (If you cannot see the issue, click 'Start over' at the right top corner of the tool.
4. ![Image](https://github.com/user-attachments/assets/812f540c-ec76-4cbb-a9bc-5c91a07205c3)

### Are you reporting an Accessibility issue?

yes

### Suggested severity

Medium - Has workaround

### Are you willing to submit a PR to fix?

no

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.

**Comment by spmonahan** [comment]
Same issue with Combobox as well: https://stackblitz.com/edit/n5f9lc-x6lebl?file=src%2Fexample.tsx

**Comment by smhigley** [comment]
So this was a bug in Accessibility Insights actually (or rather, the ruleset they use, which is axe-core). We already filed an issue on them some time ago (https://github.com/dequelabs/axe-core/issues/4625), and they had already fixed it. I can verify this fix when using the latest axe-core version to test, which does not return errors for Dropdown in your stackblitz.

The fix is in axe-core since 4.10.1, while Accessibility Insights is still on 4.9. As soon as they update, this will go away. In the meantime, I'd just ignore the error, or use axe-core directly for automated tests.
